lib/chef/knife/opc_org_list.rb

```diff
@@ -0,0 +1,45 @@
+#
+# Author:: Steven Danna (<steve@chef.io>)
+# Copyright:: Copyright 2011-2016 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+require_relative "../mixin/root_rest"
+
+module Opc
+  class OpcOrgList < Chef::Knife
+    category "CHEF ORGANIZATION MANAGEMENT"
+    banner "knife opc org list"
+
+    option :with_uri,
+      long: "--with-uri",
+      short: "-w",
+      description: "Show corresponding URIs"
+
+    option :all_orgs,
+      long: "--all-orgs",
+      short: "-a",
+      description: "Show auto-generated hidden orgs in output"
+
+    include Chef::Mixin::RootRestv0
+
+    def run
+      results = root_rest.get("organizations")
+      unless config[:all_orgs]
+        results = results.select { |k, v| !(k.length == 20 && k =~ /^[a-z]+$/) }
```

Comment:
I _think_ this filter is from when we used to pre-generate organizations because organization took a long time. I believe this is no longer the default so perhaps it would be best to turn this filtering off by default and it can end up filtering out legit organizations.